app/controllers/application_controller.rb

```diff
@@ -99,4 +99,14 @@ def after_sign_in_path_for(_)
     dashboard_path
   end
 
+  # used to deliver notification on various events
+  def notify_users(action, object_type, object_id, victims)
+    Notification.create(
+          actor: current_user,
+          action: action,
+          object_type: object_type,
+          object_id: object_id,
+          victims: victims
+        )
+  end
 end
```

Comment:
> Can we pass action as readable strings like "project_comment", "project_follow", "user_follow" etc? We can pass those strings as action argument and then use it to decide which action number do we need.
Would it be helpful in making actions more readable?

Definitely, it would. I'd suggest storing the keys in a hash in the model [like this](https://github.com/glittergallery/GlitterGallery/blob/master/app/models/issue.rb#L21-L27).

> Should we just fix it for now by adding default case in objectname, since we don't even have view for files and blobs or should we completely refactor actions, since we also need to make it readable?

Completely refactor. If we're going to support comments/actions for files/commits, we should find a way to show the user a proper string that identifies the object - It's okay if it doesn't have a view now, we can implement that later.